modules/swagger-codegen/src/main/resources/erlang-client/client_api_procession.mustache

```diff
@@ -0,0 +1,135 @@
+-module({{packageName}}_client_api_procession).
+
+-record(state, {
+    operation_id :: operation_id(),
+    validator_state :: jesse_state:state()
+}).
+
+-type operation_id() :: {{packageName}}_client_api_validation:operation_id().
+-type state() :: #state{}.
+-type params() :: #{header => any(), binding => any(), body => any(), qs_val => any()}.
+
+-export([process_request/4]).
+
+-spec process_request(Method::atom(), OperationID::operation_id(), Url::string(), Params::params()) ->
+    {ok, integer(), list(), {ok, binary()}}.
+process_request(Method, OperationID, Url, Params) ->
+    State = #state{
+        operation_id = OperationID,
+        validator_state = prepare_validator()
+    },
+
+    {PreparedUrlParams, PreparedHeaderParams, PreparedBodyParams, PreparedQueryParams} =
+        handle_request_json(Params, State),
+
+    PreparedUrl = prepare_url(Url, PreparedUrlParams, PreparedQueryParams),
+    PreparedHeaders = maps:to_list(PreparedHeaderParams),
+    PreparedBody = jsx:encode(PreparedBodyParams),
+
+    hackney:request(Method, PreparedUrl, PreparedHeaders, PreparedBody, [with_body]).
+
+-spec handle_request_json(RequestParams::params(), State::state()) ->
+    {UrlParams::map(), HeaderParams::map(), BodyParams::map(), QueryParams::map()} | {error, term()}.
```

Comment:
I would recommend to use `{ok, ....} | {error, ...}` response or `term() | no_return()`